addons/knowledge/models/knowledge_article.py

```diff
@@ -13,6 +13,51 @@
 from odoo.osv import expression
 from odoo.tools import get_lang, formataddr
 
+class ArticleFavourites(models.Model):
+    _name = 'knowledge.article.favourite'
+    _description = 'Favourite Articles'
+    _table = 'knowledge_article_favourite_user_rel'
+    _order = 'sequence asc'
+
+    article_id = fields.Many2one('knowledge.article', 'Article', required=True, ondelete='cascade')
+    user_id = fields.Many2one('res.users', 'User', index=True, required=True, ondelete='cascade')
+    sequence = fields.Integer(default=1)
+
+    def create(self, vals):
+        res = super(ArticleFavourites, self).create(vals)
+        res.reorder_favourite(res.article_id.id)
+        return res
+
+    def _set_sequence(self, article_ids):
+        for article_id in article_ids:
+            self.reorder_favourite(article_id)
```

Comment:
for each favourite, you will reorder all the favourite of the user.
So if the user has 54 favourites, you will potentially set 54 x 54 the sequences of the favourite articles.
split reorder_favourite into two part, the first one for the target article only, the second part to update all the articles the user has. When calling set_sequence from the inverse method, group the articles in self by user, then for user, call the second part. > The second part should check for duplicates, If any, resequence all the articles from that duplicated sequences. (the last updated between duplicates should not be updated)
--> Check resequence methods in knowledge_articles